feat(modeler/desktop-modeler): add details on process application in pro-code context (#6234)
camunda/camunda-docs: docs/components/modeler/desktop-modeler/process-applications.md

Desktop modeler recognizes [process applications](../../concepts/process-applications.md) you build and offers you advanced editor intelligence, deployment, and execution features within the context of such an application. To identify the boundaries of a process application, Desktop Modeler searches for a `.process-application` file in the root of your project.

> In professional software development, a typical process application contains resources such as BPMN, DMN, and Form files. These live alonside [job workers](/components/concepts/job-workers.md), implementing process logic, additional application code, and tests. How exactly your project is structured may vary depending on the implementation language, libraries, and frameworks you use.

Reviewer: [❓] Do we want to mention RPA scripts since they are supported by desktop modeler ? Since it might now fall under a typical process app and it is not part of the example I believe it is ok not to mention them.

Author: > Since it might now fall under a typical process app and it is not part of the example I believe it is ok not to mention them.

Yes, we don't have to mention everything. I think `such as` indicates that this is an open list.

## Example: Consumer loan application

Consider an application implementing consumer loan approval. It may contain:

- A main BPMN process (for example, `consumer-loan-application.bpmn`) to define the workflow.
- DMN decisions (for example, `interest-rate-calculation.dmn`, `credit-score-calculation.dmn`) for business rules.
- Forms (for example, `loan-application-review.form`) for user interactions.
- Various [job workers](/components/concepts/job-workers.md) that implement process behavior.
- Additional application code and tests

> In a standard standard Java/Maven project, such an application may be structured as follows:

Reviewer: [❓] What do you think about the following wording? If not, we should remove the second `standard`
```suggestion
In a typical Java/Maven project, the structure of such an application might be as follows:
```

└── README.md
```

## Editor support for process applications

> When you open a file in Modeler, the system implicitly determines whether it belongs to a process application. It does so by checking for the presence of a `.process-application` file in the same folder or a parent folder. Within a process application, Modeler offers advanced navigation and intelligence.

Reviewer: [☁️] Thinking out loud here. Do we have documented anywhere what `intelligence` means? I didn't find any docs around it. Might be worth linking this to a section that list some of the capabilities the editor provides in this aspect.

Author: Intelligence for example means assistance during linking, mentioned below, but will also include advanced data handling and cross-file validation in the future. Today, we offer _basic_ intelligence. But the fact that we have a search boundary (the process application root) allows us to expand and add more, in the future.

Author: ```suggestion
When you open a file in Modeler, the system implicitly determines whether it belongs to a process application. It does so by checking for the presence of a `.process-application` file in the same folder or a parent folder. Within a process application, Modeler offers improved navigation and assistance.
```

### Indicating context

Process applications are opened and closed "implicitly": A blue item in the status bar indicates whether a diagram belongs to a process application and makes all related diagrams available for navigation.

<p><img src={OverlayImg} alt="Process application" /></p>

When files of more than one process application are opened they are grouped visually.

<p><img src={GroupingImg} alt="Process application file grouping" /></p>

### Creating a process application

Create a process application by creating a `.process-application` file in the root of your project. Alternatively, create it via Modeler UI by taking the following steps:

1. Click **File > New Process Application...**.
2. Choose a folder.
3. Click **Select folder**.

> A `.process-application` file will be created in the selected folder. Any file within the folder or its subfolders will be treated as part of the process application.

Reviewer: [❓] Would it make sense to add something around the fact that the folder becomes the project root?
```suggestion
A `.process-application` file will be created in the selected folder which will now serve as the project root. 

Any file within the folder or its subfolders will be treated as part of the process application.
```
